pkg/runtime/resolver.go

```diff
@@ -59,25 +59,23 @@ type resolver struct {
 	// rules in the configuration database keyed by $namespace.
 	rules map[string][]*Rule
 
-	// Opaque ruleState used in this configuration
-	// After the resolver has no references, this state can be
-	// cleaned up correctly by the caller.
-	ruleState interface{}
-
 	// refCount tracks the number requests currently using this
 	// configuration. resolver state can be cleaned up when this count is 0.
 	refCount int32
```

Comment:
It's only modified with `atomic.AddInt32` methods, so it's fine. `atomic.Value` doesn't help us here because a read-then-write of the value is not atomic (just the access; you're guaranteed to get some consistent version of the value), which is what we need here.